azerothcore/azerothcore-wotlk
# Fix(Core/Loot): Send Retrieval Mail for lost disenchant mats

ORIGINAL AUTHOR -- Nyeriah, https://github.com/TrinityCore/TrinityCore/commit/9f606f74a3e10a275d59765312679eda9e58af99

## Changes Proposed:
- If full bags, send disenchanted materials from a group loot roll as retrieval mail
-  If auto storing loot fails, check to see whether loot type was from the 'disenchant_loot_template'. If so, send the item through item retrieval.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9256

## SOURCE:


## Tests Performed:

- disenchant first item 

## How to Test the Changes:


1. load up two wow accounts
2. make a character on each
3. .level 79 on each
4. .learn all recipes enchanting on either character (to get disenchanting)
4. group them up
5. fill bags with anything on either character
6. .go creature 79168 (Teleport to first boss of deadmines) on each
7. kill boss
8. select 'disenchant' on character with full bags, pass on other
9. .mailbox to check you have received the item

## Known Issues and TODO List:



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
